mpas_analysis/shared/generalized_reader/generalized_reader.py

```diff
@@ -171,6 +172,15 @@ def open_multifile_dataset(fileNames, calendar, config,
     # select only the data in the specified range of dates
     ds = ds.sel(Time=slice(startDate, endDate))
 
+    # limit chunk size to prevent memory error
```

Comment:
@xylar, I'm thinking that this should this be moved as a function into `mpas_xarray`'s `preprocess` with the argument `maxChunkSize` because this is a problem that needs handled for general use of xarray.  This seems better organized.